### PR TITLE
Tools page integration with api

### PR DESCRIPTION
**Pull request by vivekpipaliya**

### CODE REVIEW

File: src/components/Comment/AddComment.vue [hunk 1]
```diff
@@ -65,16 +79,29 @@ export default {
...
-        this.reset();
-        eventBus.$emit('show-toast', {body: messages.comment.commentAddSuccess, title: messages.generic.success});
-      }).catch(e => {
-        eventBus.$emit('show-toast', {body: e.message, title: messages.generic.error, type: ToastType.ERROR});
-      });
+      if (this.showRating) {
+        var payload = {
+          rating: this.rating,
+          comment: this.comment,
+          user: this.signedInUser.userName
```
Comment (souvikbasu) [inline src/components/Comment/AddComment.vue:90]: I guess it should be username

File: src/components/Comment/AddComment.vue [hunk 2]
```diff
@@ -65,16 +79,29 @@ export default {
...
+        var payload = {
+          toolId: this.toolId,
+          comment: this.$refs.commentBox.editedValue,
+          username: this.signedInUser.username
+        }
+      }
+      if(this.isEdit) {
+        this.onSave(payload, this.isEdit, this.commentId, this.commentIndex);
+      } else {
+        this.onSave(payload, this.isEdit, this.commentId, this.commentIndex);
```
Comment (souvikbasu) [inline src/components/Comment/AddComment.vue:102]: both the if and else conditions look same. What should be different in these blocks?

File: src/components/Comment/AddComment.vue [hunk 3]
```diff
@@ -65,16 +79,29 @@ export default {
...
-      });
+      if (this.showRating) {
+        var payload = {
+          rating: this.rating,
+          comment: this.comment,
+          user: this.signedInUser.userName
+        }
+      } else {
+        var payload = {
+          toolId: this.toolId,
```
Comment (souvikbasu) [inline src/components/Comment/AddComment.vue:94]: there cannot be a toolId in comment component as it is a generic component. Can we pass extra params to the payload as a param to the component and add to the payload?

File: src/components/Comment/AddComment.vue [hunk 4]
```diff
@@ -65,16 +79,29 @@ export default {
   },
   methods: {
     save() {
-      if (!this.rating) return;
+      if (this.showRating) {
+        if (!this.rating) return;
```
Comment (souvikbasu) [inline src/components/Comment/AddComment.vue:83]: use && condition in a single if block

File: src/components/Comment/AddComment.vue [hunk 5]
```diff
@@ -50,12 +51,25 @@ export default {
     showRating: {
       type: Boolean,
       required: true,
+    },
+    toolId: {
```
Comment (souvikbasu) [inline src/components/Comment/AddComment.vue:55]: toolId cannot be a prop in Comment component as it is a generic component

File: src/components/Comment/AddComment.vue (hunk 3 above)
Comment (vivekpipaliya) [reply src/components/Comment/AddComment.vue:94]: i used toolId as prop but it is not required so we can still use this component wherever we want and another thing is we cannot pass it as a param

**Pull request merged**